timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -323,11 +323,19 @@ def _construct_pandas_frame_from_data_frame(
 
         df = df.copy()
         if id_column is not None:
-            assert id_column in df.columns, f"Column {id_column} not found!"
+            assert id_column in df.columns, f"Column '{id_column}' not found!"
+            if id_column != ITEMID and ITEMID in df.columns:
+                raise ValueError(
+                    f"Data contains column '{ITEMID}'. This name is used internally by AutoGluon. Please rename this column"
+                )
```

Comment:
Is there a reason we can't rename this column for them?

Reply:
My concern was that if we quietly rename the existing column, e.g., to `item_id_` and then return predictions where the index is called `item_id` (AutoGluon internal name) the user will be confused why the values there are wrong.

However, now that I think about it again, I guess this shouldn't be a problem since this happens outside the `TimeSeriesPredictor`.